### PR TITLE
test(backend): unit tests for server.service.ts

### LINKED ISSUE #259: Test Specification — server.service.ts
## Description
Write an English-language test specification for `harmony-backend/src/services/server.service.ts` (10 functions).

## Functions to Specify
- `getPublicServers`
- `getAllServers`
- `getMemberServers`
- `getServer`
- `createServer`
- `updateServer`
- `deleteServer`
- `incrementMemberCount`
- `decrementMemberCount`
- `getMembers`

## Requirements
- List every function, its purpose, and all program paths
- Create a table with columns: **Test Purpose**, **Inputs**, **Expected Output**
- Cover happy paths, error paths (not found, duplicate slugs, unauthorized), and edge cases
- Target 80%+ code coverage of all execution paths
- Output: `docs/test-specs/server-service-spec.md`

## Acceptance Criteria
- [ ] All 10 functions listed with descriptions
- [ ] Table covers happy paths for each function
- [ ] Table covers error/exception paths
- [ ] Edge cases identified and documented
- [ ] At least one test per function

**Sprint:** 3 (P5 Testing)
**Due:** March 30, 2026

### LINKED ISSUE #264: Unit Tests — server.service.ts
## Description
Implement Jest unit tests for `harmony-backend/src/services/server.service.ts` based on the test specification from #259.

## Requirements
- Test file: `harmony-backend/tests/server.service.test.ts`
- Mock Prisma client
- Generate one test at a time via LLM to prevent hallucination
- Verify no duplicate or overlapping test cases
- Run tests locally, fix any failures

## Acceptance Criteria
- [ ] All functions from spec have at least one test
- [ ] No hallucinated test cases for non-existent functionality
- [ ] No duplicate or significantly overlapping tests
- [ ] All tests pass locally
- [ ] 80%+ code coverage (run Jest with \`--coverage\`), capture report for submission
- [ ] Tests are isolated — no cross-network calls

**Sprint:** 3 (P5 Testing)
**Due:** April 1, 2026
**Depends on:** #259

### PR DESCRIPTION
**Pull request by declanblanc**
## Summary

- Adds `harmony-backend/tests/server.service.test.ts` — 56 isolated Jest unit tests for `src/services/server.service.ts`
- All Prisma calls, `channelService`, `serverMemberService`, `isSystemAdmin`, and `eventBus` are mocked via `jest.mock`; no database or Redis connection is required
- Implements every test case from the spec in `docs/test-specs/server-service-spec.md` (closed #259)

## Coverage (scoped to `server.service.ts`)

| Metric | Result |
|---|---|
| Statements | **98.61%** |
| Branches | **97.22%** |
| Functions | **100%** |
| Lines | **98.46%** |

The single uncovered line (L82) is the `TRPCError(CONFLICT)` throw that follows the `withSlugRetry` loop — confirmed unreachable dead code, documented in the spec's edge-cases section.

## Test plan

- [x] All 56 tests pass locally (`npx jest tests/server.service.test.ts`)
- [x] No lint errors (`eslint tests/server.service.test.ts`)
- [x] Coverage ≥ 80% confirmed with `--coverage --collectCoverageFrom`
- [x] `generateSlug` — 9 normalisation cases
- [x] `getPublicServers` / `getAllServers` — filter, ordering, limit, cap-at-100
- [x] `getMemberServers` — join ordering, empty result, limit, cap
- [x] `getServer` — found and not-found paths
- [x] `createServer` — happy path, BAD_REQUEST, CONFLICT (10 candidates), P2002 retry, P2002 exhaustion (raw rethrow, not TRPCError), non-P2002 passthrough, `createDefaultChannel` failure, `addOwner` failure
- [x] `updateServer` — NOT_FOUND, FORBIDDEN, no-rename, rename+slug regen, unchanged-name (no count call), system-admin bypass, SERVER_UPDATED payload, slug exhaustion CONFLICT, withSlugRetry P2002 rethrow
- [x] `deleteServer` — NOT_FOUND, FORBIDDEN, owner, system-admin
- [x] `incrementMemberCount` — happy path + call shape
- [x] `decrementMemberCount` — happy path, null-server branch, memberCount=0 branch (both sides of compound guard)
- [x] `getMembers` — role sort, joinedAt tie-break, empty list, profile fields, unknown-role fallback

Closes #264

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: harmony-backend/tests/server.service.test.ts [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+    const error = prismaErr('P2000');
+    mockServer.count.mockResolvedValue(0);
+    mockServer.create.mockRejectedValue(error);
+
+    await expect(serverService.createServer({ name: 'New Server', ownerId: 'owner-1' }))
+      .rejects.toBeInstanceOf(Prisma.PrismaClientKnownRequestError);
+    expect(mockServer.create).toHaveBeenCalledTimes(1);
+  });
+
+  it('propagates a createDefaultChannel rejection without swallowing it', async () => {
```
Comment (acabrera04) [inline harmony-backend/tests/server.service.test.ts]: These failure-path tests only check that the error bubbles. The backend spec says `createServer` must run as a transaction across server creation, default channel creation, and owner membership creation. Since the mock Prisma client does not include `$transaction` and the test never checks rollback semantics, this suite would still pass if a downstream failure left a partially created server behind. Please add assertions that enforce that atomicity guarantee rather than only the thrown error.
